lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2633,6 +2640,29 @@ static function ( $pseudo_selector ) use ( $selector ) {
 				}
 				unset( $declarations[ $index ] );
 			}
+
+			if ( $is_root_selector && (
+					'background-size' === $declaration['name'] && 'cover' === $declaration['value'] ||
+					'background' === $declaration['name'] && ! empty( $declaration['value'] )
+				) ) {
+				$should_set_root_min_height = true;
+			}
+		}
+
+		/*
+		 * If root styles has backgroundSize === cover, or a background is set (gradient),
+		 * set the body minHeight to 100%.
+		 */
+		if ( $should_set_root_min_height ) {
+			$block_rules .= static::to_ruleset(
+				'html',
```

Comment:
Why on the HTML and not BODY?

- setting it on the body revealed potential issues, e.g., the background gradient did not always stretch (unless we used 100vh)
- Also unpredictable margins causing scrolling , e.g., an unstyled H1 at the root of the document in empty theme
- Settings "styles.dimensions.minHeight" in theme.json would overwrite the value - I suppose this would be fine for theme devs who know what they're doing, but for folks using themes that have  "styles.dimensions.minHeight" hardcoded it would mean that backgrounds don't behave as they should, that is, with the background gradient/image filling the window.
- 'HTML' is also the selector that's being used in the editor to achieve the same effect.